sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1544,17 +1650,18 @@ static void _send(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt, bool push_back)
         put_res = gnrc_netif_pktq_put(netif, pkt);
         if (put_res == 0) {
             DEBUG("gnrc_netif: (re-)queued pkt %p\n", (void *)pkt);
-            _send_queued_pkt(netif);
-            return;
+            return _send_queued_pkt(netif);
         }
         else {
             LOG_WARNING("gnrc_netif: can't queue packet for sending\n");
             /* try to send anyway */
         }
     }
-    /* hold in case device was busy to not having to rewrite *all* the link
-     * layer implementations in case `gnrc_netif_pktq` is included */
-    gnrc_pktbuf_hold(pkt, 1);
+    if (!netif->dev->driver->confirm_send) {
+        /* hold in case device was busy to not having to rewrite *all* the link
+         * layer implementations in case `gnrc_netif_pktq` is included */
+        gnrc_pktbuf_hold(pkt, 1);
```

Comment:
If we increment the reference counter of `pkt` here, do we not have to release it also in `_tx_succeeded()` and not only in `_tx_failed()`?
I am experiencing `error: packet buffer full` with `nrf24l01p_ng` when I try `ping6 fe80::a0cc:a5ff:fe6b:aa%7 -c 255`.
But when I remove `gnrc_pktbuf_hold(pkt, 1)`, the error is gone.

Reply:
Should be fixed. You tested with the upstream nrf, so no confirm_send, right?

Reply:
Yes I tested with the upstream nrf first. The release does fix it.